docs/guides/getting-started.md

```diff
@@ -0,0 +1,95 @@
+---
+page_title: Getting Started
+---
+
+# Getting Started
+
+## Installing Terraform
+
+See [this guide](https://learn.hashicorp.com/tutorials/terraform/install-cli#install-terraform) for more information.
+
+## Declaring a Task
+
+In an empty directory:
+
+1. Create a directory named `shared` to store input data and output artifacts.
+2. Create a file named `main.tf` with the following contents:
+
+```hcl
+terraform {
+  required_providers {
+    iterative = {
+      source  = "iterative/iterative"
+    }
+  }
+}
+
+provider "iterative" {}
+
+resource "iterative_task" "example" {
+  name  = "example"
+  cloud = "aws" # or any of: gcp, az, k8s
+
+  directory = "${path.root}/shared"
+
+  script = <<-END
+    #!/bin/bash
+    echo "Hello World!" > greeting.txt
+  END
+}
+```
+
+## Initializing Terraform
+
+```console
+$ terraform init
+```
+
+This command will:
+
+1. Download and install the Iterative Provider.
+2. Initialize Terraform in the current directory.
+
+~> **Note** None of the subsequent commands will work without first setting some [authentication environment variables](https://registry.terraform.io/providers/iterative/iterative/latest/docs#authentication)
+
+## Launching Tasks
+
+```console
+$ terraform apply
+···
+```
+
+This command will:
+
+1. Create all the required cloud resources.
+2. Upload the specified shared `directory` to the cloud.
```

Comment:
So is the directory `shared/` or `directory/` ?

Reply:
OK I see this now:

```hcl
directory = "${path.root}/shared"
```

So it's just the intro where both "empty directory" and `shared/` are mentioned that make this somewhat confusing.

Reply:
Perhaps `directory` without tha backticks would make it clearer?

Reply:
The confusion is that you say "given `directory`" but `directory` is not given directly, it's built from the cwd + `shared/`.

Reply:
So "given directory" sounds like you may be referring to just `shared/`.

Reply:
Maybe it's not such a big deal but if it's possible to simplify and clarify that would be great.

Reply:
`directory` is the argument name, and should be either empty/undefined (disabled) or a valid path to a directory, which will be pushed to the cloud on `apply` and pulled from the cloud on `destroy`

Reply:
`shared/` is just an example name, but `directory` can be a path to any directory in your local system.

Reply:
`directory` would be your repo for example?

Reply:
Yes, if you set `directory` to the path of your repository root, any command in the `script` will have the same behavior as locally.

Reply:
Right so I mean clarify in the doc, not just in this conversation :) as you can see it can be confusing.

Reply:
Added to https://github.com/iterative/terraform-provider-iterative/pull/237#issue-1048177975